bazel/repository_locations.bzl

```diff
@@ -1072,4 +1072,16 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         release_date = "2018-12-18",
         cpe = "N/A",
     ),
+    com_github_bufbuild_buf = dict(
+        project_name = "buf",
+        project_desc = "A new way of working with Protocol Buffers.",  # Used for breaking change detection in API protobufs
+        project_url = "https://buf.build",
+        version = "0.46.0",
+        sha256 = "4165fb44e5c0d6d5242e9f58bf53a31884a39e582abe1ce82dcb9d17c97ad54e",
+        strip_prefix = "buf",
+        urls = ["https://github.com/bufbuild/buf/releases/download/v{version}/buf-Linux-x86_64.tar.gz"],
+        release_date = "2021-07-27",
+        use_category = ["api"],
+        tags = ["manual"],
+    ),
```

Comment:
Considering that `buf` is an API dependency, should it live in `api/bazel/repository_locations.bzl`?

Reply:
OSSF Scorecard results:
```
Repo: github.com/bufbuild/buf                        
|--------|------------|-----------------------------|
| STATUS | CONFIDENCE |            NAME             |
|--------|------------|-----------------------------|
| Pass   |         10 | Active                      |
|--------|------------|-----------------------------|
| Pass   |         10 | Binary-Artifacts            |
|--------|------------|-----------------------------|
| Pass   |         10 | CI-Tests                    |
|--------|------------|-----------------------------|
| Pass   |         10 | Code-Review                 |
|--------|------------|-----------------------------|
| Pass   |         10 | Packaging                   |
|--------|------------|-----------------------------|
| Pass   |         10 | Vulnerabilities             |
|--------|------------|-----------------------------|
| Fail   |         10 | Automatic-Dependency-Update |
|--------|------------|-----------------------------|
| Fail   |          0 | Branch-Protection           |
|--------|------------|-----------------------------|
| Fail   |         10 | CII-Best-Practices          |
|--------|------------|-----------------------------|
| Fail   |         10 | Contributors                |
|--------|------------|-----------------------------|
| Fail   |         10 | Fuzzing                     |
|--------|------------|-----------------------------|
| Fail   |         10 | Pinned-Dependencies         |
|--------|------------|-----------------------------|
| Fail   |         10 | Pull-Requests               |
|--------|------------|-----------------------------|
| Fail   |         10 | SAST                        |
|--------|------------|-----------------------------|
| Fail   |         10 | Security-Policy             |
|--------|------------|-----------------------------|
| Fail   |         10 | Signed-Releases             |
|--------|------------|-----------------------------|
| Fail   |         10 | Signed-Tags                 |
|--------|------------|-----------------------------|
| Fail   |         10 | Token-Permissions           |
|--------|------------|-----------------------------|
```

Reply:
moved to api/bazel

Reply:
@yaseenalk can you tag the Buf maintainers you have been working with on this thread? I think we should ask them to cleanup the failed OSSF scorecard checks, since these are good security hygiene. It doesn't have to block the use of this here, but one of hte goals of our use of Scorecard is that we spot relatively easy to rectify problems and fix them.

Reply:
@bufdev

Reply:
I ran the OSSF scorecard tool locally to go though the issues, here's some notes. An overall note is that github.com/bufbuild/buf ("public") is a partial replica of our internal monorepo ("internal"), so there are some mechanisms scored via OSSF that make less sense in this context. Note that we will likely be changing this mechanism, making github.com/bufbuild/buf the source of truth for the code it owns, but right now, we copy from internal to public.

- Automatic-Dependency-Update: We don't have dependabot enabled as we have our own mechanism to make sure that every time we copy from internal to public, all dependencies are updated. So we do continually update public for every new commit to it.
- Branch-Protection: We have branch protection enabled on `main`, however not every rule that would make sense for a non-replica repository. I've just turned on linear history, but (1) we don't require reviews pre-merge, as the code has been reviewed in internal (2) we don't require conversations to be resolved for the same reason.
- CII-Best-Practices: We don't have this badge, I don't see a reason for us to add it right now, but I assume this isn't an issue.
- Code-Review: We do full code review on every commit, the review is just done on internal, and not on public.
- Contributors: Almost all of our development work is done by Buf employees/contractors.
- Fuzzing: We haven't set this up yet, but can in the future.
- Pinned-Dependencies: We do pin all of our dependencies for the codebase. The warnings printed are for GitHub Actions jobs, and false positive for Docker images that it says are unpinned. For example, `golang:1.16.6-alpine3.4` is detected as unpinned - technically sure, we aren't using a hash here, but this is a pretty reliable/readable mechanism to make sure we're pinned to a specific micro for Golang.
- SAST: Filed https://github.com/bufbuild/buf/issues/395
- Security-Policy: Any recommendations here for a good security policy to pattern off of would be great.
- Signed-Releases: Filed https://github.com/bufbuild/buf/issues/396
- Signed-Tags: Filed https://github.com/bufbuild/buf/issues/396
- Token-Permissions: Filed https://github.com/bufbuild/buf/issues/397

Reply:
Thanks for your thoughtful and detailed response. The raised issues are great and with the addition of a `SECURITY.md` security policy + the follow ups I think you will be in good shape. There is a good article covering `SECURITY.md` best practices which includes links to good examples at https://snyk.io/blog/ten-git-hub-security-best-practices/#4-add-a-securitymd-file